devicetypes/fibargroup/fibaro-door-window-sensor-2.src/fibaro-door-window-sensor-2.groovy

```diff
@@ -0,0 +1,516 @@
+		sendEvent(name: "multiStatus", value: statusValue, descriptionText: statusValue, displayed: display)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	logging("${device.displayName} woke up", "info")
+	def cmds = []
+	sendEvent(descriptionText: "$device.displayName woke up", isStateChange: true)
+	if ( state.wakeUpInterval?.state == "notSynced" && state.wakeUpInterval?.value != null ) {
+		cmds << zwave.wakeUpV2.wakeUpIntervalSet(seconds: state.wakeUpInterval.value as Integer, nodeid: zwaveHubNodeId)
+		state.wakeUpInterval.state = "synced"
+	}
+	cmds << zwave.batteryV1.batteryGet()
+	cmds << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1)
+	runIn(1,"syncNext")
+	[response(encapSequence(cmds,1000))]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def paramKey = parameterMap().find( {it.num == cmd.parameterNumber } ).key
+	logging("${device.displayName} - Parameter ${paramKey} value is ${cmd.scaledConfigurationValue} expected " + state."$paramKey".value, "info")
+	state."$paramKey".state = (state."$paramKey".value == cmd.scaledConfigurationValue) ? "synced" : "incorrect"
+	syncNext()
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationRejectedRequest cmd) {
+	logging("${device.displayName} - rejected request!","warn")
+	for ( param in parameterMap() ) {
+		if ( state."$param.key"?.state == "inProgress" ) {
+			state."$param.key"?.state = "failed"
+			break
+		} 
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.alarmv2.AlarmReport cmd) {
+	def map = [:]
+	logging("${device.displayName} - AlarmReport received, zwaveAlarmType: ${cmd.zwaveAlarmType}, zwaveAlarmEvent: ${cmd.zwaveAlarmEvent}", "info")
+	switch (cmd.zwaveAlarmType) {
+		case 6:
+			map.name = "contact"
+			switch (cmd.zwaveAlarmEvent) {
+				case 22:
+					map.value = "open"
+					map.descriptionText = "${device.displayName} is open"
+                    break
+				case 23:
+					map.value = "closed"
+					map.descriptionText = "${device.displayName} is closed"
+					break
+			}
+			break
+		case 7:
+			map.name = "tamper"
+			switch (cmd.zwaveAlarmEvent) {
+				case 0:
+					map.value = "clear"
+					map.descriptionText = "Tamper alert cleared"
+					break
+				case 3:
+					map.value = "detected"
+					map.descriptionText = "Tamper alert: sensor removed or covering opened"
+					break
+			}
+			break
+		case 4:
+			if (device.currentValue("temperatureAlarm")?.value != null) {
+				map.name = "temperatureAlarm"
+				switch (cmd.zwaveAlarmEvent) {
+					case 0:
+						map.value = "clear"
+						map.descriptionText = "Tamper alert cleared"
+						break
+					case 2:
+						map.value = "overheat"
+						map.descriptionText = "Tamper alert: sensor removed or covering opened"
+						break
+					case 6:
+						map.value = "underheat"
+						map.descriptionText = "Tamper alert: sensor removed or covering opened"

```

Comment:
All the descriptionText values for temperature alarms are wrong

Reply:
Ha! Good catch! Good ol' copy and paste...